paper/src/main/java/com/denizenscript/denizen/paper/utilities/PaperAPIToolsImpl.java

```diff
@@ -414,4 +412,17 @@ public void setMaterialTags(Material type, Set<NamespacedKey> tags) {
     public double[] getRecentTps() {
         return Bukkit.getTPS();
     }
+
+    @Override
+    public String getCopperGolemState(CopperGolem copperGolem) {
+        return copperGolem.getWeatheringState().name();
+    }
+
+    @Override
+    public void setCopperGolemState(ElementTag variant, CopperGolem copperGolem) {
+        WeatheringCopperState state = variant.asEnum(WeatheringCopperState.class);
+        if (state != null) {

```

Comment:
You'd want to use `Mechanism#requireEnum` here, otherwise there'd be no error (you can just pass both the mechanism and the input to this method).
Beyond that LGTM.